sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -895,4 +895,27 @@ class AnalysisSuite extends AnalysisTest with Matchers {
     assertAnalysisError(testRelation2.select(RowNumber() + 1),
       Seq("Window function row_number() requires an OVER clause."))
   }
+
+  test("SPARK-32237: Hint in CTE") {
```

Comment:
Could you add end-2-end tests, too, e.g., in `SQLQueryTestSuite` or `SQLQuerySuite`?